### PR TITLE
refactor(Guild): Destructure object in guild editing

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
This style of code isn't consistent with other parts of the library. I think destructuring the object here looks pretty neat and seems to result in code that's easier to look at. Personally, every time I've come across this, I think about doing this but end up not bothering. But now's the time, apparently!

I also reordered the parameters as they are in Discord's documentation.

There was also a pretty suspicious max-len rule modification here which I removed. It must have served an ideal purpose in the past, but presently, it doesn't seem to do much at all.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by SpaceEEC — APPROVED** [review]

File: packages/discord.js/src/structures/Guild.js [hunk 1]
```diff
@@ -788,51 +787,59 @@ class Guild extends AnonymousGuild {
...
+    banner,
+    systemChannel,
+    systemChannelFlags,
+    rulesChannel,
+    publicUpdatesChannel,
+    preferredLocale,
+    features,
+    description,
+    premiumProgressBarEnabled,
+    reason,
```
Comment (kyranet) [inline packages/discord.js/src/structures/Guild.js:810]: There are some fields that are as-is with Discord's, we can rest those an map only the camelCase values (and `reason`).

Those would be `name`, `features`, and `description`.

File: packages/discord.js/src/structures/Guild.js [hunk 2]
```diff
@@ -788,51 +787,59 @@ class Guild extends AnonymousGuild {
...
+  }) {
+    const {
+      client: { actions, channels, rest, users },
+      id,
+    } = this;
```
Comment (kyranet) [inline packages/discord.js/src/structures/Guild.js:815]: ngl I'm not a fan of this, there are fields that can be lazily called below, and fields that are always called, but nevertheless, we don't do this pattern anywhere else.

**Review by kyranet — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/Guild.js (hunk 2 above)
Comment (Jiralite) [reply packages/discord.js/src/structures/Guild.js]: Reverted in 5e088a929b3b151cf4a8aaa3e46f705c5bd85abe.

File: packages/discord.js/src/structures/Guild.js (hunk 1 above)
Comment (Jiralite) [reply packages/discord.js/src/structures/Guild.js]: Should be done with f9dcd8d5745d2a2cc9ec715b284dc7180799606d.

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**